src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -97,14 +97,35 @@ HeapWord* ShenandoahFreeSet::allocate_single(ShenandoahAllocRequest& req, bool&
 
   // Overwrite with non-zero (non-NULL) values only if necessary for allocation bookkeeping.
 
+  bool allow_new_region = true;
+  switch (req.affiliation()) {
+    case ShenandoahRegionAffiliation::OLD_GENERATION:
+      if (_heap->old_generation()->adjusted_unaffiliated_regions() <= 0) {
```

Comment:
This is really a test for ==, and the compiler doesn't complain because the test is meaningful as written (though perhaps confusing as written).  OTOH, writing it this way makes the code more "future proof" in case someone changes the return type to signed.